control/init.sh

```diff
@@ -17,14 +17,21 @@ function realpath() {
 
 
 if [ -z ${CI_TEST} ]; then
-    if [ ! -f $VENV/bin/activate ]; then
-        if [[ $CCHQ_VIRTUALENV == *3.10* ]]; then
-          # use venv because 3.10 setup includes installing python3.10-venv
-          python3.10 -m venv $VENV
+    if hash python3.10 2>/dev/null && [[ $( source /etc/os-release; echo $VERSION_ID ) == 18.04 ]]; then
+        # if on 18.04 with 3.10 installed, use cchq-3.10
+        CCHQ_VIRTUALENV=$CCHQ_VIRTUALENV-3.10
+        VENV=~/.virtualenvs/$CCHQ_VIRTUALENV
+    fi
+    # check if a virtualenv at $VENV exists yet, and create if not
+    if [[ ! -f $VENV/bin/activate ]]; then
+        if hash python3.10 2>/dev/null; then
```

Comment:
Replace the first line of this file with:
```py
if hash python3.10 2>/dev/null && [[ $( source /etc/os-release; echo $VERSION_ID ) == 18.04 ]]; then
    # if on 18.04 with 3.10 installed, use cchq-3.10
    CCHQ_VIRTUALENV=${CCHQ_VIRTUALENV:-cchq-3.10}
else
    CCHQ_VIRTUALENV=${CCHQ_VIRTUALENV:-cchq}
fi
```

And apply this:
```suggestion
    # check if a virtualenv at $VENV exists yet, and create if not
    if [[ ! -f $VENV/bin/activate ]]; then
        if [[ "$CCHQ_VIRTUALENV" =~ -3\.10$ ]]; then
```

This way arbitrary custom 3.10 environments can be created by using a suffix of `-3.10` on the name. Virtualenv names not ending with `-3.10` will continue to use the default system python3.

Reply:
Hmm I am surprised we have to activate the virtualenv there. I was under the impression that the init script would be run after logging into the machine which would activate the script. Though this seems to have stemmed from https://github.com/dimagi/commcare-cloud/commit/4725f8130a4e78882075ff3b2a6be1482f9f6c43 which aimed to optimize this. @dannyroberts what do you think of partially reverting this by running `source ~/init-ansible` instead of `source ~/.virtualenvs/${{CCHQ_VIRTUALENV:-cchq}}/bin/activate` to allow better control over the virtualenv used?

Reply:
One reason for virtualenv activation vs running the init script is to skip the things in the init script that take a long time (updating pip and ansible galaxy requirements, etc.). It's a performance optimization that is very valuable when all you want to do is ssh to a machine, for example.

Reply:
So alternatively, we could just rely on an environment variable being set on the control machine right? I know it isn't the best approach because we lose control, but we could log a message to the user saying they are overriding commcare-cloud's default behavior and that they should remove the variable (assuming we haven't encouraged custom virtualenv names in the past).